homeassistant/components/edl21/sensor.py

```diff
@@ -368,13 +375,16 @@ def event(self, message_body) -> None:
             else:
                 entity_description = SENSORS.get(obis)
                 if entity_description and entity_description.name:
-                    name = entity_description.name
-                    if self._name:
-                        name = f"{self._name}: {name}"
-
+                    device_name = (
+                        self._name or DEFAULT_DEVICE_NAME
+                    )  # self._name is only used for backwards compatibility of sensors from configuration.yaml and will be removed
```

Comment:
Please place the comment above the code, and split to avoid long lines.
```suggestion
                    # self._name is only used for backwards YAML compatibility
                    # This needs to be cleaned up when YAML support is removed
                    device_name = self._name or DEFAULT_DEVICE_NAME
```